app/controllers/comments_controller.rb

```diff
@@ -2,4 +2,28 @@ class CommentsController < ApplicationController
   def index
     @comments = Comment.all
   end
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @user = User.find(params[:user_id])
+    @comment = Comment.new(comment_params)
+    @comment.post = @post
+    @comment.author = @user
+
+    if @comment.save
+      redirect_to "/users/#{@user.id}/posts"
+    else
+      render 'posts/show'
+    end
+  end
```

Comment:
- [ ] Comments should be created on behalf of the `current_user`, not the user from the URL param. Kindly fix this in your create method.